src/appliance-onboarding-script/appliance_setup/avs/avsarconboarder/orchestrator/network/_network_orchestrator.py

```diff
@@ -0,0 +1,37 @@
+from ...orchestrator._orchestrator import Orchestrator
+import ipaddress
+from avs.avsarconboarder.exception import InvalidInputError
+
+class NetworkOrchestrator(Orchestrator):
+    def __init__(self):
+        self.list_of_ip_addr_in_segment = list()
+
+    def validate_static_ip_cidr_block(self, config):
+        segmant_ip_cidr = config["staticIpNetworkDetails"]["networkCIDRForApplianceVM"]
+        gateway_ip      = ""
+        char_index      = 0
+        while char_index < len(segmant_ip_cidr) and segmant_ip_cidr[char_index] != '/':
+            gateway_ip += segmant_ip_cidr[char_index]
+            char_index += 1
+
+        cidr = segmant_ip_cidr[char_index - len(segmant_ip_cidr) + 1:]
+        if int(cidr) != 28:
+            raise InvalidInputError("Invalid segamnt block size provided, Please provide a /28 address")
```

Comment:
>InvalidInputError [](http://example.com/codeflow?start=18&length=17)

What is the behavior if host bits are set? The description lists it as an unsupported scenario. Where are checking for this?

Reply:
We are handling this exception as explained in the above comment
https://github.com/Azure/ArcOnAVS/pull/42#discussion_r978773560

Reply:
checking it in validate function